File: src/Application Specific/Classes/Entity Collections/ContractsAlternative/ContractAlternativeSiteJobItems/ContractAlternativeSiteJobItemsValidator.vb
Namespace Entity

    Namespace ContractAlternativeSiteJobItems

        Public Class ContractAlternativeSiteJobItemsValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oContractAlternativeSiteJobItems As ContractAlternativeSiteJobItems)

                'make sure that contact object is valid
                If oContractAlternativeSiteJobItems.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oContractAlternativeSiteJobItems.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If oContractAlternativeSiteJobItems.VisitFrequencyID = 0 Then
                    AddCriticalMessage("Visit Frequency Missing")
                End If


                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/ContractsAlternative/ContractAlternativeSiteJobItems/ContractAlternativeSiteJobItemsValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ContractAlternativeSiteJobItems
    {
        // make sure that contact object is valid
        public class ContractAlternativeSiteJobItemsValidator : BaseValidator
        {
            public void Validate(ContractAlternativeSiteJobItems oContractAlternativeSiteJobItems)
            {
                // make sure that contact object is valid
                if (oContractAlternativeSiteJobItems.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oContractAlternativeSiteJobItems.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oContractAlternativeSiteJobItems.VisitFrequencyID == 0)
                {
                    AddCriticalMessage("Visit Frequency Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}